### PR TITLE
[PL-38164] Add delegate selector details

### PR DESCRIPTION
**Pull request by brian-f-harness**
Add delegate selector details

For reviewers, previews are available:
[Add a custom remote script and manifests](https://64ac10c145395b007afb75b0--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/add-a-custom-remote-script-and-manifests/)
[Using shell scripts in CD stages](https://64ac10c145395b007afb75b0--harness-developer.netlify.app/docs/continuous-delivery/x-platform-cd-features/executions/cd-general-steps/using-shell-scripts/)
[Set up a local runner build infrastructure](https://64adbe6a3e41d140467a9407--harness-developer.netlify.app/docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure/)
[Set up a Kubernetes cluster build infrastructure](https://64adbe6a3e41d140467a9407--harness-developer.netlify.app/docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure/)
[Select delegates with selectors](https://64adbe6a3e41d140467a9407--harness-developer.netlify.app/docs/first-gen/firstgen-platform/account/manage-delegates/select-delegates-for-specific-tasks-with-selectors/)
[Harness Delegate FAQs](https://64ac10c145395b007afb75b0--harness-developer.netlify.app/docs/frequently-asked-questions/harness-faqs/harness-delegate-faqs/)
[Delegate overview](https://64ac10c145395b007afb75b0--harness-developer.netlify.app/docs/platform/Delegates/delegate-concepts/delegate-overview)
[Use delegate selectors](https://64ac10c145395b007afb75b0--harness-developer.netlify.app/docs/platform/delegates/manage-delegates/select-delegates-with-selectors/)
[Install Harness Delegate on Kubernetes or Docker](https://64ac10c145395b007afb75b0--harness-developer.netlify.app/tutorials/platform/install-delegate/)

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by pratmit — APPROVED** [review]
LGTM

**Review by jennyjames21 — DISMISSED** [review]

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md [hunk 1]
```diff
@@ -110,18 +110,22 @@ For instructions on creating clusters, go to:
 
 :::tip
 
-Although you must select a specific delegate when you create the Kubernetes Cluster connector, you can choose to use a different delegate for executions and cleanups in individual pipelines or stages. To do this, use [pipeline-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#option-pipeline-delegate-selector) or [stage-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#option-stage-delegate-selector).
+Although you must select a specific delegate when you create the Kubernetes Cluster connector, you can choose to use a different delegate for executions and cleanups in individual pipelines, steps, or stages. To do this, use [pipeline-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#pipeline-delegate-selector), [step and step group-level selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#step-and-step-group-delegate-selector), or [stage-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#stage-delegate-selector).
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md:113]: Please revert the changes to this file. CI doesn't support step-level selectors - See info box here: https://64adbe6a3e41d140467a9407--harness-developer.netlify.app/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#step-and-step-group-delegate-selector

**Review by michaelcretzman — DISMISSED** [review]
aside from the broken partials, the rest of the PR looks fine.
I will approve and you can just fix the partials before merging.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md [hunk 2]
```diff
@@ -110,18 +110,22 @@ For instructions on creating clusters, go to:
...
-2. Pipeline
-3. Connector
+1. Step
+2. Step Group
+3. Stage
+4. Pipeline
+5. Connector
 
-This means that if delegate selectors are present at the pipeline and stage levels, then these selections override the delegate selected in the Kubernetes Cluster connector's configuration. If a stage has a stage-level delegate selector, then it uses that delegate. Stages that don't have stage-level delegate selectors use the pipeline-level selector, if present, or the connector's delegate.
+This means that if delegate selectors are present at the pipeline, step, or stage levels, then these selections override the delegate selected in the Kubernetes cluster connector's configuration. If a stage has a stage-level delegate selector, then it uses that delegate. Stages that don't have stage-level delegate selectors use the pipeline-level selector, if present, or the connector's delegate.
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md:123]:
```suggestion
This means that if delegate selectors are present at the pipeline and stage levels, then these selections override the delegate selected in the Kubernetes cluster connector's configuration. If a stage has a stage-level delegate selector, then it uses that delegate. Stages that don't have stage-level delegate selectors use the pipeline-level selector, if present, or the connector's delegate.
```

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md [hunk 3]
```diff
@@ -110,18 +110,22 @@ For instructions on creating clusters, go to:
...
 
 For example, assume you have a pipeline with three stages called `alpha`, `beta`, and `gamma`. If you specify a stage-level delegate selector on `alpha` and you don't specify a pipeline-level delegate selector, then `alpha` uses the stage-level delegate, and the other stages (`beta` and `gamma`) use the Connector delegate.
 
+Delegate selectors override service infrastructure connectors when present at the pipeline, step, or stage levels.
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md:127]:
```suggestion
Delegate selectors override service infrastructure connectors when present at the pipeline or stage levels.
```

I don't know what a service infrastructure connector is. Do you mean the delegate in the build infrastructure connector (i.e. k8s cluster connector)? I believe I already described that in the section/example above this, but if not, could you put a link here to more info of what a "service infra connector" is?

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 4]
```diff
@@ -294,18 +294,23 @@ In the pipeline's build stage (`type: CI`), insert `platform` and `runtime` spec
 
 :::tip
 
-Although you must install a delegate to use the local runner build infrastructure, you can choose to use a different delegate for executions and cleanups in individual pipelines or stages. To do this, use [pipeline-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#option-pipeline-delegate-selector) or [stage-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#option-stage-delegate-selector).
+Although you must install a delegate to use the local runner build infrastructure, you can choose to use a different delegate for executions and cleanups in individual pipelines, steps, or stages. To do this, use [pipeline-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#pipeline-delegate-selector), [step and step group-level selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#step-and-step-group-delegate-selector), or [stage-level delegate selectors](/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#stage-delegate-selector).
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:297]: Pls revert - CI doesn't support step-level selectors. See info box here: https://64adbe6a3e41d140467a9407--harness-developer.netlify.app/docs/platform/Delegates/manage-delegates/select-delegates-with-selectors#step-and-step-group-delegate-selector

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 5]
```diff
@@ -294,18 +294,23 @@ In the pipeline's build stage (`type: CI`), insert `platform` and `runtime` spec
...
 
 Delegate selections take precedence in the following order:
 
-1. Stage
-2. Pipeline
-3. Platform (build machine delegate)
+1. Step
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:301]: Pls revert.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 6]
```diff
@@ -294,18 +294,23 @@ In the pipeline's build stage (`type: CI`), insert `platform` and `runtime` spec
...
+3. Stage
+4. Pipeline
+5. Connector
+6. Platform (build machine delegate)
 
 This means that if delegate selectors are present at the pipeline and stage levels, then these selections override the platform delegate, which is the delegate that you installed on the build machine. If a stage has a stage-level delegate selector, then it uses that delegate. Stages that don't have stage-level delegate selectors use the pipeline-level selector, if present, or the platform delegate.
 
 For example, assume you have a pipeline with three stages called `alpha`, `beta`, and `gamma`. If you specify a stage-level delegate selector on `alpha` and you don't specify a pipeline-level delegate selector, then `alpha` uses the stage-level delegate, and the other stages (`beta` and `gamma`) use the platform delegate.
 
+Delegate selectors do not override service infrastructure connectors. Delegate selectors only determine the delegate that executes the operations of your pipeline.
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:312]:
```suggestion
Delegate selectors do not override service infrastructure connectors. Delegate selectors only determine the delegate that executes the operations of your pipeline.
```

Confused why this statement is different than the one on the K8s build infra page. Also, same comment as on the k8s page - Idk what a "service infrastructure connector" is.

**Review by aimurphy — CHANGES_REQUESTED** [review]
CI doesn't support step/step group selectors.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md (hunk 6 above)
Comment (brian-f-harness) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:312]: removed

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md (hunk 5 above)
Comment (brian-f-harness) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:301]: reverted

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md (hunk 4 above)
Comment (brian-f-harness) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:297]: reverted. only left updated cross reference change (broken in prod) - updated to remove option from both xrefs

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md (hunk 2 above)
Comment (brian-f-harness) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md:123]: reverted

**Review by aimurphy — APPROVED** [review]

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md (hunk 1 above)
Comment (brian-f-harness) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure.md]: only update I left in place was to correct the xref (removed "option")

**Pull request merged**